de.metas.contracts/src/main/sql/postgresql/system/50-de.metas.contracts/5494490_sys_gh4085_InvoiceCandidateRefundRelation.sql

```diff
@@ -0,0 +1,62 @@
+;
+
+-- 2018-05-24T18:18:02.556
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_Reference_Trl (AD_Language,AD_Reference_ID, Description,Help,Name, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy) 
+SELECT l.AD_Language,t.AD_Reference_ID, t.Description,t.Help,t.Name, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy FROM AD_Language l, AD_Reference t 
+WHERE l.IsActive='Y' AND l.IsSystemLanguage='Y' AND l.IsBaseLanguage='N' AND t.AD_Reference_ID=540867 AND NOT EXISTS (SELECT 1 FROM AD_Reference_Trl tt WHERE tt.AD_Language=l.AD_Language 
+AND tt.AD_Reference_ID=t.AD_Reference_ID)
+;
+
+-- 2018-05-24T18:19:33.708
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_Ref_Table (AD_Client_ID,AD_Display,AD_Key,AD_Org_ID,AD_Reference_ID,AD_Table_ID,AD_Window_ID,Created,CreatedBy,EntityType,IsActive,IsValueDisplayed,Updated,UpdatedBy)
+ VALUES (0,545845,545845,0,540867,540321,540092,TO_TIMESTAMP('2018-05-24 18:19:33','YYYY-MM-DD HH24:MI:SS'),100,'de.metas.contracts','Y','N',TO_TIMESTAMP('2018-05-24 18:19:33','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2018-05-24T18:24:19.810
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_Ref_Table SET WhereClause=' exists (  select 1 from C_Invoice_Candidate refund  join C_Invoice_Candidate_Assignment ica on refund.C_Invoice_Candidate_ID = ica.C_Invoice_Candidate_Term_ID  join C_Invoice_Candidate normal on ica.C_Invoice_Candidate_Assigned_ID  where normal.C_Invoice_Candidate_ID = @C_Invoice_Candidate_ID/-1@ and refund.C_Invoice_Candidate_ID = C_Invoice_Candidate.C_Invoice_Candidate_ID  )',Updated=TO_TIMESTAMP('2018-05-24 18:24:19','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Reference_ID=540867
+;
+
+-- 2018-05-24T18:24:43.326
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_RelationType SET AD_Reference_Target_ID=540867,Updated=TO_TIMESTAMP('2018-05-24 18:24:43','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_RelationType_ID=540204
+;
+
+-- 2018-05-24T18:24:51.734
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_RelationType SET IsDirected='Y',Updated=TO_TIMESTAMP('2018-05-24 18:24:51','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_RelationType_ID=540204
+;
+
+
+-- 2018-05-25T10:27:35.321
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_Ref_Table SET WhereClause=' exists (  select 1 from C_Invoice_Candidate refund  join C_Invoice_Candidate_Assignment ica on refund.C_Invoice_Candidate_ID = ica.C_Invoice_Candidate_Term_ID  join C_Invoice_Candidate normal on ica.C_Invoice_Candidate_Assigned_ID = normal.C_Invoice_Candidate_ID  where normal.C_Invoice_Candidate_ID = @C_Invoice_Candidate_ID/-1@ and refund.C_Invoice_Candidate_ID = C_Invoice_Candidate.C_Invoice_Candidate_ID  )',Updated=TO_TIMESTAMP('2018-05-25 10:27:35','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Reference_ID=540867
```

Comment:
wondering if we can't skip the `C_Invoice_Candidate refund` after all and do
```sql
exists ( 
	select 1 from C_Invoice_Candidate_Assignment ica 
	join C_Invoice_Candidate normal on ica.C_Invoice_Candidate_Assigned_ID = normal.C_Invoice_Candidate_ID  
	where normal.C_Invoice_Candidate_ID = @C_Invoice_Candidate_ID/-1@ and ica.C_Invoice_Candidate_Term_ID = C_Invoice_Candidate.C_Invoice_Candidate_ID
)
```
wdyt?

Reply:
I think it would be fine. changing it now and trying with my data.